cmd/main.go

```diff
@@ -0,0 +1,260 @@
+
+	data := &keptnevents.ProblemEventData{}
+	if err := event.DataAs(data); err != nil {
+		logger.Error(fmt.Sprintf("Got Data Error: %s", err.Error()))
+		return err
+	}
+
+	if event.Type() != keptnevents.ProblemOpenEventType {
+		const errorMsg = "Received unexpected keptn event"
+		logger.Error(errorMsg)
+		return errors.New(errorMsg)
+	}
+
+	go createEvent(event, shkeptncontext, *data, logger)
+
+	return nil
+}
+
+func createEvent(event cloudevents.Event, shkeptncontext string, data keptnevents.ProblemEventData, logger *keptnutils.Logger) error {
+
+	kubeClient, err := keptnutils.GetKubeAPI(true)
+
+	if err != nil {
+		logger.Error("could not create Kubernetes client")
+		return errors.New("could not create Kubernetes client")
+	}
+
+	snowAPIURL, snowUser, snowPassword, err := utils.GetGlobalServicenowCredentials(kubeClient, logger)
+
+	if err != nil {
+		logger.Debug(err.Error())
+		logger.Debug("Failed to fetch global ServiceNow credentials, exiting.")
+		return err
+	}
+
+	var EventSource string = "KEPTN"
+	var KeptnImpactedEntity = strings.Split(data.ImpactedEntity, " ")
+	var KeptnNode string = KeptnImpactedEntity[len(KeptnImpactedEntity)-1]
+
+	var problemInfo problemDetailsData
+	ProblemData := []byte(data.ProblemDetails)
+	Source := (*json.RawMessage)(&ProblemData)
+	err = json.Unmarshal(*Source, &problemInfo)
+	if err != nil {
+		panic(err)
+	}
+
+	// Map eventtype to ServiceNow event severity
+	var severity string = ""
+	// Map problem status to ServiceNow event resolution state
+	var resolutionState = ""
+
+	switch problemInfo.SeverityLevel {
+	case "AVAILABILITY":
+		severity = "1"
+	case "ERROR":
+		severity = "2"
+	case "PERFORMANCE":
+		severity = "3"
+	case "RESOURCE_CONTENTION":
+		severity = "4"
+	case "CUSTOM_ALERT":
+		severity = "3"
+	default:
+		severity = "4"
+	}
+
+	if problemInfo.Status == "RESOLVED" {
+		resolutionState = "Closing"
+	} else {
+		resolutionState = "New"
+	}
+
+	SnowEvent := Event{
+		Records: []Records{
+			Records{
+				Source:          EventSource,
+				EventClass:      event.Source(),
+				Node:            KeptnNode,
+				MetricName:      data.ProblemTitle,
+				Type:            data.ProblemTitle,
+				MessageKey:      data.PID,
+				Severity:        severity,
+				Description:     data.ProblemTitle,
+				ResolutionState: resolutionState,
+				AdditionalInfo:  "{'KeptnSource':'" + event.Source() + "', 'KeptnContext':'" + shkeptncontext + "', 'KeptnImpactedEntity':'" + KeptnNode + "', 'KeptnPID':'" + data.PID + "', 'KeptnProblemTitle':'" + data.ProblemTitle + "', 'KeptnProblemState':'" + data.State + "', 'KeptnProject':'" + data.Project + "', 'KeptnStage':'" + data.Stage + "', 'KeptnService':'" + data.Service + "', 'KeptnProblemID':'" + data.ProblemID + "', 'KeptnImpactLevel':'" + problemInfo.ImpactLevel + "', 'KeptnSeverityLevel':'" + problemInfo.SeverityLevel + "'}",
+			},
+		},
+	}
+
+	var EventData []byte
+	EventData, err = json.Marshal(SnowEvent)
+	if err != nil {
+		log.Println(err)
```

Comment:
please always use the keptn go-utils logger (`logger.Debug(err.Error())`, instead of `log.Println` or `fmt.Println`

Reply:
added commit to use keptn go-utils logger for error logging